### PR TITLE
CIV-12722 Notifications for updated HWF number

### PR DESCRIPTION
**Pull request by paul-pearson**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-12722

### Change description ###

Notifications for updated HWF number

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/UpdateHelpWithFeeRefNumberHandlerTest.java [hunk 1]
```diff
@@ -58,6 +60,8 @@ void shouldUpdateHwFReferenceNumberSuccessfully_FeeType_ClaimIssued() {
             CaseData data = getCaseData(response);
             Assertions.assertThat(data.getCaseDataLiP().getHelpWithFees().getHelpWithFeesReferenceNumber()).isEqualTo("7890");
             Assertions.assertThat(data.getClaimIssuedHwfDetails().getHwfReferenceNumber()).isNull();
+            Assertions.assertThat(data.getBusinessProcess().getCamundaEvent()).isEqualTo(NOTIFY_LIP_CLAIMANT_HWF_OUTCOME.toString());
```
Comment (manish14836) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/UpdateHelpWithFeeRefNumberHandlerTest.java:63]: Assertions keyword in not required in each line as it is statically imported above. Remove from all places

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyLiPClaimantHwFOutcomeHandlerTest.java [hunk 2]
```diff
@@ -131,12 +135,76 @@ void shouldNotifyApplicant_HwfOutcome_NoRemission_Hearing() {
...
+        @Test
+        void shouldNotifyApplicant_HwfOutcome_RefNumberUpdated_Hearing() {
+            // Given
+            HelpWithFeesDetails hwfeeDetails = HelpWithFeesDetails.builder()
+                .hwfCaseEvent(UPDATE_HELP_WITH_FEE_NUMBER)
+                .build();
+            CaseData caseData = CaseDataBuilder.builder().atStateClaimSubmitted().build().toBuilder()
+                .applicant1(PartyBuilder.builder().individual().build().toBuilder()
+                                .partyEmail(EMAIL)
+                                .build())
```
Comment (manish14836) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyLiPClaimantHwFOutcomeHandlerTest.java:185]: Reuse Case Data from as created in recent changes. Please check Pablo's test cases.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyLiPClaimantHwFOutcomeHandlerTest.java [hunk 3]
```diff
@@ -131,12 +135,76 @@ void shouldNotifyApplicant_HwfOutcome_NoRemission_Hearing() {
...
             );
         }
 
+        @Test
+        void shouldNotifyApplicant_HwfOutcome_RefNumberUpdated_ClaimIssued() {
+            // Given
+            HelpWithFeesDetails hwfeeDetails = HelpWithFeesDetails.builder()
+                .hwfCaseEvent(UPDATE_HELP_WITH_FEE_NUMBER)
+                .build();
+            CaseData caseData = CaseDataBuilder.builder().atStateClaimSubmitted().build().toBuilder()
```
Comment (manish14836) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyLiPClaimantHwFOutcomeHandlerTest.java:150]: Reuse Case Data from as created in recent changes. Please check Pablo's test cases.

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimantClaimMatcher.java [hunk 4]
```diff
@@ -315,4 +315,9 @@ public boolean isPaymentPlanRejected() {
     public boolean isHwfNoRemission() {
         return caseData.isHWFOutcomeReady() && caseData.getHwFEvent() == CaseEvent.NO_REMISSION_HWF;
     }
+
+    @Override
+    public boolean isHwfUpdatedRefNumber() {
```
Comment (manish14836) [inline src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimantClaimMatcher.java:320]: Add test case for this

Comment (paul-pearson) [reply src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimantClaimMatcher.java:320]: Done

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/UpdateHelpWithFeeRefNumberHandlerTest.java (hunk 1 above)
Comment (paul-pearson) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/UpdateHelpWithFeeRefNumberHandlerTest.java]: Done

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyLiPClaimantHwFOutcomeHandlerTest.java (hunk 2 above)
Comment (paul-pearson) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyLiPClaimantHwFOutcomeHandlerTest.java]: Done

**Review by manish14836 — APPROVED** [review]

**Review by ElifDenizEsea — APPROVED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Pull request merged**